plugins/modules/cloud/misc/proxmox_interfaces.py

```diff
@@ -0,0 +1,414 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Andreas Botzner (@botzner_andreas) <andreas at botzner dot com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: proxmox_interfaces
+short_description: Management network interfaces on a Proxmox node.
+version_added: 3.5.0
+description:
+  - Allows you to create/update/delete network interfaces of a Proxmox node.
+author: "Andreas Botzner (@botzner_andreas) <andreas at botzner dot com>"
+options:
+  config:
+    description: Interface configuration
+    required: true
+    type: list
+    elements: dict
+    suboptions:
+      name:
+        description:
+          - Name of the network interface.
+        type: str
+        required: true
+      type:
+        description:
+          - Network interface type
+        type: str
+        default: bridge
+        choices:
+        - bridge
+        - bond
+        - eth
+        - alias
+        - vlan
+        - OVSBridge
+        - OVSBond
+        - OVSPort
+        - OVSIntPort
+        - unknown
+      address:
+        description:
+          - IPv4 address of the interface
+        type: str
+      address6:
+        description:
+          - IPv6 address of the interface
+        type: str
+      autostart:
+        description:
+          - Automatically start interface on boot.
+        type: bool
+        default: true
+      bond_primary:
+        description:
+          - Primary interface for active-backup bond.
+        type: str
+      bond_mode:
+        description:
+          - Bonding mode
+        type: str
+        choices:
+          - balance-rr
+          - active-backup
+          - balance-xor
+          - broadcast
+          - 802.3ad
+          - balance-tlb
+          - balance-alb
+          - balance-slb
+          - lacp-balance-slb
+          - lacp-balance-tcp
+      bond_xmit_hash_policy:
+        description:
+          - Selects the transmit hash policy to use for slave selection in balance-xor and 802.3ad
+        type: str
+        choices:
+          - layer2
+          - layer2+3
+          - layer3+4
+      bridge_ports:
+        description:
+          - Specify the interfaces you want to add to your bridge.
+        type: str
+      bridge_vlan_ports:
+        description:
+          - Enable bridge vlan support.
+        type: bool
+      cidr:
+        description:
+          - IPv4 CIDR
+        type: str
+      cidr6:
+        description:
+          - IPv6 CIDR
+        type: str
+      comments:
+        description:
+          - Comments
```

Comment:
The description doesn't help me understanding what's the difference between `comments` and `comments6`.

Reply:
I'm not to sure what comments6 does either. The API documentation only says 'comments' as a description.
I tried to the Proxmox API with both the `comments` and `comments6` argument. It created the following entry in `/etc/network/interfaces`: 
```
iface vmbr135 inet manual
	bridge-ports none
	bridge-stp off
	bridge-fd 0
	comments6 THIS IS A COMMENTS6
#THIS IS A COMMENT
```
The `comments` option results in a simple comment in the file and the comments6 option is a config value. Though I am not sure what's the use for it since the `comments6` comment didn't show up in the PVE GUI interface, while the normal comment shows up next to the interface.